amqp_connection.c

```diff
@@ -459,6 +465,17 @@ static PHP_METHOD(amqp_connection_class, __construct)
 		}
 	}
 
+	zend_update_property_double(this_ce, getThis(), ZEND_STRL("rpc_timeout"), INI_FLT("amqp.rpc_timeout") TSRMLS_CC);
+
+	if (ini_arr && PHP5to7_ZEND_HASH_FIND(HASH_OF(ini_arr), "rpc_timeout", sizeof("rpc_timeout"), zdata)) {
```

Comment:
I've tried various DEFAULT_RPC_TIMEOUT values with rpc_timeout defined neither in config, nor in associative arrays and ""/"0" produced the same result with hanging, and for example "5" made a 5-second wait before exit in poll(). Maybe i should define "0" instead of "", and i will fix that.

Tested cases:
* Defaults, no mention of rpc_timeout anywhere - described hereinabove
* Defined only in php.ini as amqp.rpc_timeout - works as expected. 0 hangs, >0 waits and returns
* Defined only in associative array passed to AMQPConnection constructor in php - same result as above

But are you pointing out other problem? I don't really understand

Reply:
Sounds good, let’s go with "0" instead

Reply:
Hello. Done in https://github.com/pdezwart/php-amqp/pull/334/commits/1838f9915d60e33ac1e9f700ec72427c115a8af3